tests/unit/test_routes.py

```diff
@@ -371,7 +373,122 @@ def test_update_resource(module_client, module_db, fake_auth_from_oc):
     assert (response.status_code == 404)
 
 
-def test_bad_requests(module_client, module_db, fake_auth_from_oc):
+def test_search(module_client, module_db, fake_auth_from_oc, fake_algolia_save, fake_algolia_search):
+    client = module_client
+
+    first_term = ''.join(random.choices(string.ascii_uppercase + string.digits, k=10))
+    apikey = get_api_key(client)
+
+    # Create resource and find it in the search results.
+    resource = client.post("/api/v1/resources",
+                          json=dict(
+                              name=f"{first_term}",
+                              category="Website",
+                              url=f"{first_term}",
+                              paid=False,
+                          ),
+                          headers={'x-apikey': apikey}
+                          )
+    result = client.get(f"/api/v1/search?q={first_term}")
+
+    assert (resource.status_code == 200)
+    assert (result.status_code == 200)
+    assert (result.json['data'][0]['url'] == resource.json['data'].get('url'))
+
+    # Update the resource and test that search results reflect changes
+    updated_term = ''.join(random.choices(string.ascii_uppercase + string.digits, k=10))
+    resource_id = resource.json['data'].get('id')
+    resource = client.put(f"/api/v1/resources/{resource_id}",
+                           json=dict(
+                               url=f"{updated_term}",
+                           ),
+                           headers={'x-apikey': apikey}
+                           )
+
+    result = client.get(f"/api/v1/search?q={updated_term}")
+
+    assert (resource.status_code == 200)
+    assert (result.status_code == 200)
+    assert (result.json['data'][0]['url'] == resource.json['data'].get('url'))
+
+
+def test_algolia_exception_error(module_client, module_db, fake_auth_from_oc, fake_algolia_exception):
+    client = module_client
+    first_term = ''.join(random.choices(string.ascii_uppercase + string.digits, k=10))
+    apikey = get_api_key(client)
+
+    result = client.get(f"/api/v1/search?q=python")
+
+    assert (result.status_code == 500)
+
+    resource = client.post("/api/v1/resources",
```

Comment:
I'm not sure I understand what the purpose is behind this API call and the next one for `put`. We already have a `test_create_resource` and a `test_update_resource` that create and update a resource and check that the status code is 200.

Is the purpose to demonstrate that even when algolia throws an exception, we still return a 200? I think we should also test that something else happens to notify us that this is going on, since we're going to want to fix it but we don't want the Resources API to go down just because algolia is down